File: M3 Intermediate DSA/src/L6_Arrays_Carry_Forward/Assignment4.java
package L6_Arrays_Carry_Forward;

public class Assignment4 {

  public long solve(String A){
    int count  = 0;
    long ans = 0;
    int N = A.length();

    for (int index = N -1; index >= 0; index--) {

      if (A.charAt(index) == 'G') {
        count++;
      }

      if (A.charAt(index) == 'A') {
        ans += count;
      }
    }

    return ans;
  }

}

File: M3_Intermediate_DSA/L4_Carry_Forward/assignment4.go
package main

func SolveA4(A string) int64 {
	var ans int64
	count := 0

	for _, ch := range A {
		switch ch {
		case 'A':
			count++
		case 'G':
			ans += int64(count)
		}
	}

	return ans
}
